tests/system/src/test/java/test/robot/javafx/scene/SRGBTest.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.scene;
+
+import javafx.geometry.Bounds;
+import javafx.geometry.Point2D;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.Scene;
+import javafx.scene.image.PixelReader;
+import javafx.scene.image.WritableImage;
+import javafx.scene.layout.HBox;
+import javafx.scene.paint.Color;
+import javafx.scene.robot.Robot;
+import javafx.scene.shape.Rectangle;
+import javafx.stage.Screen;
+import javafx.stage.Stage;
+
+import java.awt.color.ColorSpace;
+import java.awt.GraphicsEnvironment;
+import javax.swing.SwingUtilities;
+
+import com.sun.glass.ui.Application;
+import com.sun.glass.ui.Window;
+import com.sun.javafx.PlatformUtil;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.Test;
+import static org.junit.Assume.assumeTrue;
+
+import test.robot.testharness.VisualTestBase;
+
+public class SRGBTest extends VisualTestBase {
+
+    private static final int SWATCH_SIZE = 200;
+
+    // Avoid testing colors out near the edge of the gamut to avoid clipping
+    // when converting between spaces.
+    static final float LOW = 0.25f;
+    static final float MID = 0.50f;
+    static final float HIGH = 0.75f;
+
+    // The component tolerance allows one bit of rounding when writing a color
+    // out and another bit when reading it back in.
+    static final double COMPONENT_TOLERANCE = 2.0 / 255.0;
```

Comment:
suggestion: 2.0001 / 255.0 to account for floating point errors

Reply:
I'm not sure this test needs to be that precise. And I'm fine with the threshold falling below 2.0/255.0 since it should be as small as possible (I even experimented briefly with lower numbers). I haven't run into an issue on the three platforms I've run this test on.

Reply:
Then the comment might not be technically correct, because it is possible to have "two bit difference" in the integer color values, yet fail due to small error intrinsic in floating point operations.

Example:
```
a=253 b=251 delta=0.007843137254902044 tol=0.00784313725490196

delta=0.007843137254902044
  tol=0.00784313725490196  (2.0 / 255.0)
```

Reply:
I've updated the constant as you suggested.